### PR TITLE
templates:.*etcd-member.yaml wait for etcd membership to be updated.

### PR DESCRIPTION
**Pull request by alaypatel07**

### CODE REVIEW

File: templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml [hunk 1]
```diff
@@ -77,6 +77,32 @@ contents:
...
+            echo "waiting for $ETCD_DNS_NAME to be added to etcd membership"
+            sleep 2
+          done
+        terminationMessagePolicy: FallbackToLogsOnError
+        volumeMounts:
+        - name: discovery
+          mountPath: /run/etcd/
+        - name: certs
+          mountPath: /etc/ssl/etcd/
+        - name: sa
```
Comment (alaypatel07) [inline templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml]: dont really need this mount. I'll remove it shortly

File: templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml [hunk 2]
```diff
@@ -77,6 +77,32 @@ contents:
...
+        - -c
+        - |
+          #!/bin/sh
+          set -euxo pipefail
+
+          source /run/etcd/environment
+
+          export ETCDCTL_API=3 ETCDCTL_CACERT=/etc/ssl/etcd/ca.crt ETCDCTL_CERT=$(find /etc/ssl/ -name *peer*crt) ETCDCTL_KEY=$(find /etc/ssl/ -name *peer*key)
+
+          until $(etcdctl member list | grep $ETCD_DNS_NAME>>/dev/null)
```
Comment (hexfusion) [inline templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml]: note: if you want to send command output to /dev/null you can use a single `>` operator

File: templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml [hunk 3]
```diff
@@ -77,6 +77,32 @@ contents:
           mountPath: /etc/ssl/etcd/
         - name: sa
           mountPath: /var/run/secrets/kubernetes.io/serviceaccount/
+      - name: wait-for-membership
```
Comment (hexfusion) [inline templates/master/00-master/_base/files/etc-kubernetes-manifests-etcd-member.yaml]: this is kind of long how about just `membership` or `scale` or ? less to type :)

**Review by hexfusion — COMMENTED** [review]
looks good added a few comments. Also, I wonder how we get out of this state. Meaning if for some reason we never scale. Should we set a hard limit? I think it's fine for now but worth considering how we cover maybe in observation. Thanks!

**Comment by alaypatel07** [comment]
if we don't scale, would it go through without the restart controller? If we don't scale would it be fair to say that we need the pod to die?

**Pull request merged**